main.c

```diff
@@ -871,20 +924,44 @@ init_kms(struct vkcube *vc)
 
       insert_vk_chain(&img_info, &external_img_info);
 
-      VkSubresourceLayout plane_layouts[1] = {
-         {
+      int plane_count = 0;
+      VkSubresourceLayout plane_layouts[2];
+      if(is_linear) {
+         plane_count = 1;
+         plane_layouts[0] = (VkSubresourceLayout){
             .offset = offset,
             .rowPitch = stride,
             .arrayPitch = 0,
             .depthPitch = 0,
             .size = 0,
-         },
-      };
+         };
+      } else {
+         plane_count = 2;
```

Comment:
The plane count is `VkDrmFormatModifierPropertiesEXT::drmFormatModifierPlaneCount`, and can be anywhere from 1 to 4.